src/main/java/jarvis/task/Task.java

```diff
@@ -0,0 +1,152 @@
+package jarvis.task;
+
+import java.util.Scanner;
+
+import jarvis.duration.Duration;
+import jarvis.exception.command.CommandParseException;
+import jarvis.exception.command.MissingParameterException;
+
+/**
+ * Abstract task class.
+ */
+public abstract class Task {
+    private boolean isDone;
+    private final String description;
+
+    /**
+     * Constructor for a task, marked as undone.
+     *
+     * @param description Description of the task.
+     * @throws MissingParameterException If description is null or blank.
+     */
+    public Task(String description) throws MissingParameterException {
+        if (description == null || description.isBlank()) {
+            throw new MissingParameterException("Missing description", "A task description is needed.");
+        }
+
+        this.isDone = false;
+        this.description = description;
+    }
+
+    /**
+     * Constructor for a task.
+     *
+     * @param description Description of the task.
+     * @param isDone Whether the task is marked as done.
+     * @throws MissingParameterException If description is null or blank.
+     */
+    public Task(String description, boolean isDone) throws MissingParameterException {
+        this(description);
+        this.isDone = isDone;
+    }
+
+    /**
+     * Serializes a task for storage.
+     *
+     * @return Serial string.
+     */
+    public abstract String serialize();
+
+    /**
+     * Creates a task from the given serial string.
+     *
+     * @param serial Serial string.
+     * @return Created task.
+     */
+    public static Task deserialize(String serial) {
+        if (serial == null || serial.isBlank()) {
+            return null;
+        }
+
+        Scanner scanner = new Scanner(serial).useDelimiter("\\s*/\\s*");
+        String type = "";
+        boolean isDone = false;
+        String description = null;
+        String deadline = null;
+        String fromDateTime = null;
+        String toDateTime = null;
+        String duration = null;
+
+        if (scanner.hasNext()) {
+            type = scanner.next();
+        }
+        if (scanner.hasNextBoolean()) {
+            isDone = scanner.nextBoolean();
+        }
+        if (scanner.hasNext()) {
+            description = scanner.next();
+        }
+
+        try {
+            switch (type) {
+            case ToDoTask.ID:
```

Comment:
Nice use of constants in these `switch-case` statements! They are very descriptive when paired with the class name that they originate from.